internal/workflow_testsuite.go

```diff
@@ -827,6 +972,27 @@ func (e *TestWorkflowEnvironment) SetOnLocalActivityCanceledListener(
 	return e
 }
 
+func (e *TestWorkflowEnvironment) SetOnNexusOperationStartedListener(
+	listener func(service string, operation string, input converter.EncodedValue),
+) *TestWorkflowEnvironment {
+	e.impl.onNexusOperationStartedListener = listener
+	return e
+}
+
+func (e *TestWorkflowEnvironment) SetOnNexusOperationCompletedListener(
+	listener func(service string, operation string, result converter.EncodedValue, err error),
+) *TestWorkflowEnvironment {
+	e.impl.onNexusOperationCompletedListener = listener
+	return e
+}
+
+func (e *TestWorkflowEnvironment) SetOnNexusOperationCanceledListener(
```

Comment:
nit: do we have a test for `SetOnNexusOperationCanceledListener`?

Reply:
No, I'm not sure how to test it since the SDK itself doesn't support canceling a Nexus operation.

Reply:
You can cancel a nexus operation from the SDK, the SDK can send a cancelation command from a workflow task.